pkg/nodeconfig/nodeconfig.go

```diff
@@ -265,6 +299,35 @@ func (nc *nodeConfig) applyLabelsAndAnnotations(labels, annotations map[string]s
 	return nil
 }
 
+// configureAzureCloudNodeManager configures azure cloud node manager
+func (nc *nodeConfig) configureAzureCloudNodeManager() error {
+	cfg, err := crclientcfg.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, "unable to get config to talk to kubernetes api server")
+	}
+
+	client, err := clientset.NewForConfig(cfg)
+	if err != nil {
+		return errors.Wrap(err, "unable to get client from the given config")
+	}
+
+	// We need to deploy Cloud Node Manager on Azure if Cloud Controller is owned by Cloud Controller Manager
+	if nc.platformType == configv1.AzurePlatformType {
+		ownedByCCM, err := cluster.IsCloudControllerOwnedByCCM(client)
+		if err != nil {
+			return err
+		}
+		if ownedByCCM {
+			// Configuring cloud node manager for Azure platform
+			if err := nc.Windows.ConfigureAzureCloudNodeManager(nc.node.GetName()); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
```

Comment:
This can be restructured to return early if not azure, will reduce indentation and increase readability.
```
 func (nc *nodeConfig) configureAzureCloudNodeManager() error {
     if nc.platformType != configv1.AzurePlatformType {
       return nil
     }
     ...
 }
```